src/libraries/System.Private.CoreLib/src/Microsoft/Win32/SafeHandles/SafeFileHandle.Windows.cs

```diff
@@ -288,7 +289,31 @@ unsafe long GetFileLengthCore()
 
                 if (!Interop.Kernel32.GetFileInformationByHandleEx(this, Interop.Kernel32.FileStandardInfo, &info, (uint)sizeof(Interop.Kernel32.FILE_STANDARD_INFO)))
                 {
-                    throw Win32Marshal.GetExceptionForLastWin32Error(Path);
+                    if (!this._path?.StartsWith(@"\\?\", StringComparison.Ordinal) ?? true)
+                    {
+                        throw Win32Marshal.GetExceptionForLastWin32Error(Path);
+                    }
+
+                    byte[] buffer = ArrayPool<byte>.Shared.Rent(sizeof(Interop.Kernel32.StorageReadCapacity));
```

Comment:
stackalloc?

Reply:
The P/Invoke accepts a byte array. We would need a new overload. And not even `stackalloc` would be needed than, just declaring a struct as local and passing its address.

Reply:
Is stackalloc better?

I was trying to be consistent with existing code.

https://github.com/dotnet/runtime/blob/a5f3676cc71e176084f0f7f1f6beeecd86fbeafc/src/libraries/System.Private.CoreLib/src/System/IO/FileSystem.Windows.cs#L574